### ISSUE TITLE
"high"-mark not updating

### ISSUE THREAD

**Issue by robbobfrh84** [matthewmain/kiss_the_sky#79]
... I wonder if that Save / Resume could be interrupting this somehow... 

I Actually saved this game a few times. And, I know after saving it I got a new High mark. However, now it won't updating the 56% on this current game I have. 

<img width="731" alt="Screen Shot 2019-05-04 at 3 32 44 PM" src="https://user-images.githubusercontent.com/15821138/57184497-f6aed400-6e81-11e9-94ba-3b77b360e425.png">

**Comment by robbobfrh84** [comment]
This could be a clue. I played a game a long time and it'd passed the 50% mark years previous. However, when I clicked the game button to resume the game it gave me this message again...? 


<img width="669" alt="Screen Shot 2019-05-04 at 7 04 58 PM" src="https://user-images.githubusercontent.com/15821138/57186164-b5c5b800-6e9f-11e9-89cb-49f07acc8254.png">

**Comment by matthewmain** [comment]
Weird. I'll dig into this and fix it. (Also, probably a related issue, right now if you're playing in "ambient mode", then resume a "game mode" game, the game stays in ambient mode, and vice versa. Will fix this too while going through the game save/resume code.)

**Comment by robbobfrh84** [comment]
oh ya... I forgot that I actually hard-coded "game mode" in the `resumeState()` function trying to get the landing to go away. But, I now realize that that data is in the saved object and can be selected by that condition. On the react side we could just pass that variable as an argument to the `resumeState()` func?

**Comment by matthewmain** [comment]
So after looking at this, I think what's going on in the first screenshot is that the flowers aren't technically "red" flowers, which is why the percentage/marker isn't updating. One of the challenges with making the game was defining a "red" flower based on HSL values. There's always going to be a shade of dark pink or orange that's on the edge of the arbitrary values I settled on for a "red" flower. Which can be annoying, because sometimes flowers that look _really_ close to red don't get counted as red. I tried to account for this by adding those pulsing red halos you see around flowers that are "officially" red so there's as little confusion as possible, but there are probably always going to be edge cases of reddish flowers that don't quite qualify. 

The second screenshot is a legit bug cased by my `resumeSavedGame()` function not updating the UI for game mode versus ambient mode, so the previous mode carries over after the game is resumed and confuses the code that tracks game progress. [This PR](https://github.com/matthewmain/kiss_the_sky/pull/80) should fix this by now properly resetting the mode in `resumeState()`. I've tested this through a few games/saves and as far as I can tell it's working consistently. So I'll close this issue for now, but if you run into any similar bugs down the road open it back up and I'll have another look.
